apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/BaseSparkApp.java

```diff
@@ -35,29 +34,39 @@
 @Slf4j
 public abstract class BaseSparkApp {
   private static final long HEARTBEAT_INTERVAL_SECONDS_DEFAULT = 300;
-  protected static final Meter METER = OtelConfig.getMeter(BaseSparkApp.class.getName());
+  protected static final String METRICS_SCOPE = BaseSparkApp.class.getName();
   protected final String jobId;
   protected final StateManager stateManager;
   private final long heartbeatIntervalSeconds;
+  protected final OtelEmitter otelEmitter;
   private final ScheduledExecutorService scheduledExecutorService =
       Executors.newSingleThreadScheduledExecutor();
 
-  protected BaseSparkApp(String jobId, StateManager stateManager) {
-    this(jobId, stateManager, HEARTBEAT_INTERVAL_SECONDS_DEFAULT);
+  protected BaseSparkApp(String jobId, StateManager stateManager, OtelEmitter otelEmitter) {
+    this(jobId, stateManager, HEARTBEAT_INTERVAL_SECONDS_DEFAULT, otelEmitter);
   }
 
   protected BaseSparkApp(String jobId, StateManager stateManager, long heartbeatIntervalSeconds) {
+    this(jobId, stateManager, heartbeatIntervalSeconds, AppsOtelEmitter.getInstance());
```

Comment:
So this will initialize default oss instance of otel. How do customize this and pass additional params for new fluent bit model emission?

Reply:
This is the constructor will OtelEmitter is not provided, we give it a default value.